src/transformers/models/omdet_turbo/modeling_omdet_turbo.py

```diff
@@ -0,0 +1,1916 @@
+            feat_low = fpn_states[-1]
+            feat_high = fpn_feature_maps[idx + 1]
+            downsample_feat = self.downsample_convs[idx](feat_low)
+            hidden_states = self.pan_blocks[idx](
+                torch.concat([downsample_feat, feat_high.to(downsample_feat.device)], dim=1)
+            )
+            fpn_states.append(hidden_states)
+
+        if not return_dict:
+            return tuple(v for v in [fpn_states, encoder_states, all_attentions] if v is not None)
+        return BaseModelOutput(last_hidden_state=fpn_states, hidden_states=encoder_states, attentions=all_attentions)
+
+
+class OmDetTurboMLPWithDropout(nn.Module):
+    def __init__(self, d_input, d_output, d_hidden=1024, dropout=0.1, activation="relu"):
+        super(OmDetTurboMLPWithDropout, self).__init__()
+        self.linear1 = nn.Linear(d_input, d_hidden)
+        self.activation = ACT2FN[activation]
+        self.dropout = nn.Dropout(dropout)
+        self.linear2 = nn.Linear(d_hidden, d_output)
+
+    def forward(self, x):
+        return self.linear2(self.dropout(self.activation(self.linear1(x))))
+
+
+class OmDetTurboMLP(nn.Module):
+    """Very simple multi-layer perceptron (also called FFN)"""
+
+    def __init__(self, input_dim, hidden_dim, output_dim, num_layers):
+        super().__init__()
+        self.num_layers = num_layers
+        h = [hidden_dim] * (num_layers - 1)
+        self.layers = nn.ModuleList(nn.Linear(n, k) for n, k in zip([input_dim] + h, h + [output_dim]))
+
+    def forward(self, x):
+        for i, layer in enumerate(self.layers):
+            x = F.relu(layer(x)) if i < self.num_layers - 1 else layer(x)
+        return x
+
+
+class OmDetTurboResidualLayer(nn.Module):
+    """
+    A residual connection followed by a layer norm.
+    """
+
+    def __init__(self, size, dropout):
+        super().__init__()
+        self.norm1 = nn.LayerNorm(size)
+        self.dropout = nn.Dropout(dropout)
+
+    def forward(self, x, y):
+        "Apply residual connection to any sublayer with the same size."
+        return self.norm1(x + self.dropout(y))
+
+
+class OmDetTurboResidualMLP(nn.Module):
+    def __init__(self, d_m, dropout, d_hidden=1024, activation="relu"):
+        super().__init__()
+        self.mlp = OmDetTurboMLPWithDropout(d_m, d_m, d_hidden, dropout, activation)
+        self.res1 = OmDetTurboResidualLayer(d_m, dropout)
+
+    def forward(self, x):
+        mlp_out = self.mlp(x)
+        x = self.res1(x, mlp_out)
+        return x
+
+
+class OmDetTurboDeformableTransformerDecoderLayer(nn.Module):
+    """
+    https://github.com/PaddlePaddle/PaddleDetection/blob/develop/ppdet/modeling/transformers/deformable_transformer.py
+    https://github.com/fundamentalvision/Deformable-DETR/blob/main/models/deformable_transformer.py
+    """
+
+    def __init__(
+        self,
+        config,
+        encoder_hidden_dim=256,
+        n_heads=8,
+        d_ffn=1024,
+        dropout=0.0,
+        act=nn.ReLU(),
```

Comment:
We shouldn't have an instance as a default argument to a method because of binding. Instead, we can have this as `None` and default to `nn.ReLU()`. The standard pattern in the library is to allow people to specify the activation through the config as string or activation function and use that to set the value